src/main/java/org/primefaces/util/MapBuilder.java

```diff
@@ -45,7 +45,7 @@ private MapBuilder(Map<K, V> map) {
     }
 
     public static <K, V> MapBuilder<K, V> builder() {
-        return new MapBuilder<>(new HashMap<K, V>());
+        return new MapBuilder<>(new LinkedHashMap<K, V>());
```

Comment:
@Rapster any harm in making the default MapBuilder always use a LinkedHashMap so the values are always stored in the order they are put in the map?

Reply:
LinkedHashMap will take up more memory, I don't think it should be default. Instead introducing two 2 methods hash and linked would be more practical, and remove builder methods

Reply:
OK I created two methods `hashed()` and `linked()` but I left `builder()` just call hashed() for backwards compatibility